src/components/views/dialogs/DeviceVerifyDialog.js

```diff
@@ -21,58 +22,261 @@ import MatrixClientPeg from '../../../MatrixClientPeg';
+        const VerificationShowSas = sdk.getComponent('views.verification.VerificationShowSas');
+        return <VerificationShowSas
+            sas={this._showSasEvent.sas}
+            onCancel={this._onCancelClick}
+            onDone={this._onSasMatchesClick}
+        />;
+    }
+
+    _renderSasVerificationPhaseWaitForPartnerToConfirm() {
+        const Spinner = sdk.getComponent('views.elements.Spinner');
+        return <div>
+            <Spinner />
+            <p>{_t(
+                "Waiting for %(userId)s to confirm...", {userId: this.props.userId},
+            )}</p>
+        </div>;
+    }
+
+    _renderSasVerificationPhaseVerified() {
+        const VerificationComplete = sdk.getComponent('views.verification.VerificationComplete');
+        return <VerificationComplete onDone={this._onVerifiedDoneClick} />;
+    }
+
+    _renderSasVerificationPhaseCancelled() {
+        const VerificationCancelled = sdk.getComponent('views.verification.VerificationCancelled');
+        return <VerificationCancelled onDone={this._onCancelClick} />;
+    }
+
+    _renderLegacyVerification() {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        const AccessibleButton = sdk.getComponent('views.elements.AccessibleButton');
+
+        const key = FormattingUtils.formatCryptoKey(this.props.device.getFingerprint());
+        const body = (
+            <div>
+                <AccessibleButton
+                    element="span" className="mx_linkButton" onClick={this._onSwitchToSasClick}
+                >
+                    {_t("Use two-way text verification")}
+                </AccessibleButton>
+                <p>
+                    { _t("To verify that this device can be trusted, please contact its " +
+                        "owner using some other means (e.g. in person or a phone call) " +
+                        "and ask them whether the key they see in their User Settings " +
+                        "for this device matches the key below:") }
+                </p>
+                <div className="mx_UserSettings_cryptoSection">
+                    <ul>
+                        <li><label>{ _t("Device name") }:</label> <span>{ this.props.device.getDisplayName() }</span></li>
+                        <li><label>{ _t("Device ID") }:</label> <span><code>{ this.props.device.deviceId }</code></span></li>
+                        <li><label>{ _t("Device key") }:</label> <span><code><b>{ key }</b></code></span></li>
+                    </ul>
+                </div>
+                <p>
+                    { _t("If it matches, press the verify button below. " +
+                        "If it doesn't, then someone else is intercepting this device " +
+                        "and you probably want to press the blacklist button instead.") }
+                </p>
+                <p>
+                    { _t("In future this verification process will be more sophisticated.") }
```

Comment:
Yeah, wasn't sure about this since you'll still just get this screen if you don't have the labs flag turned on. Maybe still better to remove it though.